ee/models/session_recording_extensions.py

```diff
@@ -65,41 +63,62 @@ def persist_recording(recording_id: str, team_id: int) -> None:
-        "version": "2022-12-22",
-        "distinct_id": recording.distinct_id,
-        "snapshot_data_by_window_id": recording.snapshot_data_by_window_id,
-    }
-
-    # TODO: This is a hack workaround for datetime conversion
-    string_content = json.dumps(content, default=str)
-    analytics_payload["content_size_in_bytes"] = len(string_content.encode("utf-8"))
-    string_content = compress_to_string(string_content)
-    analytics_payload["compressed_size_in_bytes"] = len(string_content.encode("utf-8"))
-
-    logger.info("Persisting recording: writing to S3...", recording_id=recording_id, team_id=team_id)
-
-    try:
-        object_path = recording.build_object_storage_path()
-        object_storage.write(object_path, string_content.encode("utf-8"))
-        recording.object_storage_path = object_path
-        recording.save()
-
-        analytics_payload["total_time_ms"] = (timezone.now() - start_time).total_seconds() * 1000
-        report_team_action(recording.team, "session recording persisted", analytics_payload)
+    # if snapshots are already in blob storage, then we can just copy the files between buckets
+    with SNAPSHOT_PERSIST_TIME_HISTOGRAM.labels(source="S3").time():
+        target_prefix = recording.build_object_storage_path()
+        source_prefix = recording.build_blob_ingestion_storage_path()
+        copied_count = object_storage.copy_objects(source_prefix, target_prefix)
 
-        logger.info("Persisting recording: done!", recording_id=recording_id, team_id=team_id)
-    except object_storage.ObjectStorageError as ose:
-        capture_exception(ose)
-        report_team_action(recording.team, "session recording persist failed", analytics_payload)
-        logger.error(
-            "session_recording.object-storage-error", recording_id=recording.session_id, exception=ose, exc_info=True
-        )
+    if copied_count > 0:
+        recording.storage_version = "2023-08-01"
+        recording.object_storage_path = target_prefix
+        recording.save()
+        logger.info("Persisting recording: done!", recording_id=recording_id, team_id=team_id, source="s3")
+        return
+    else:
+        # TODO this can be removed when we're happy with the new storage version
+        with SNAPSHOT_PERSIST_TIME_HISTOGRAM.labels(source="ClickHouse").time():
+            recording.load_snapshots(100_000)  # TODO: Paginate rather than hardcode a limit
+
+            content: PersistedRecordingV1 = {
+                "version": "2022-12-22",
+                "distinct_id": recording.distinct_id,
+                "snapshot_data_by_window_id": recording.snapshot_data_by_window_id,
+            }
+
+            string_content = json.dumps(content, default=str)
+            string_content = compress_to_string(string_content)
+
+            logger.info("Persisting recording: writing to S3...", recording_id=recording_id, team_id=team_id)
+
+            try:
+                object_path = recording.build_object_storage_path()
```

Comment:
Doesn't this mean for the legacy style "/data" will now be a file instead of a folder? I feel like this will pop up for us at some point where we will be like "why does the list call fail??" and we won't have any way of easily debugging whats up...

I know its a pain but I feel like we should _keep_ the old values in the old format... That way if we do ever try and do  a migration or something it will be straight forward to know where they are.